src/main/java/au/org/ala/biocache/dao/SearchDAOImpl.java

```diff
@@ -911,120 +1054,118 @@ public void writeCoordinatesToStream(SearchRequestParams searchParams,OutputStre
                     sensitiveQ.addAll(splitQueries(queries, dd.getSensitiveFq(), sensitiveSOLRHdr, notSensitiveSOLRHdr));
                 }
     
-                //multi-thread the requests...
-                ExecutorService pool = Executors.newFixedThreadPool(6);
-                Set<Future<Integer>> futures = new HashSet<Future<Integer>>();
+                //Set<Future<Integer>> futures = new HashSet<Future<Integer>>();
                 final AtomicInteger resultsCount = new AtomicInteger(0);
                 final boolean threadCheckLimit = checkLimit;
     
-                //execute each query, writing the results to stream
+                List<Callable<Integer>> solrCallables = new ArrayList<>(queries.size());
+                // execute each query, writing the results to stream
                 for(final SolrQuery splitByFacetQuery: queries){
-                    //define a thread
+                    // define a thread
                     Callable<Integer> solrCallable = new Callable<Integer>(){
-    
-                        int startIndex = 0;
-    
                         @Override
                         public Integer call() throws Exception {
+                            int startIndex = 0;
+                            // Randomise the wakeup time so they don't all wakeup on a periodic cycle
+                            long localThrottle = throttle + Math.round(Math.random() * throttle);
+        
                             String [] fq = downloadParams.getFq();
                             if (splitByFacetQuery.getFilterQueries() != null && splitByFacetQuery.getFilterQueries().length > 0) {
-                                if (fq == null) fq = new String [0];
+                                if (fq == null) {
+                                    fq = new String[splitByFacetQuery.getFilterQueries().length];
```

Comment:
Sorry, this change was done assuming the semantics was the same as the JDK utils that reuse the array if it is big enough. Will revert as commons-lang doesn't seem to reuse this array.